Build failure: nixos-rebuild-ng

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix build nixpkgs#nixos-rebuild-ng`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/302140208

### Relevant log output

```console
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
unpacking source archive /nix/store/dmbmbqrrrxavk6qx0g4xdzv7awxyylnk-src
source root is src
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "src/tests/test_utils.py"
Running phase: patchPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
running egg_info
creating nixos_rebuild_ng.egg-info
writing nixos_rebuild_ng.egg-info/PKG-INFO
writing dependency_links to nixos_rebuild_ng.egg-info/dependency_links.txt
writing entry points to nixos_rebuild_ng.egg-info/entry_points.txt
writing top-level names to nixos_rebuild_ng.egg-info/top_level.txt
writing manifest file 'nixos_rebuild_ng.egg-info/SOURCES.txt'
reading manifest file 'nixos_rebuild_ng.egg-info/SOURCES.txt'
writing manifest file 'nixos_rebuild_ng.egg-info/SOURCES.txt'
* Building wheel...
running bdist_wheel
running build
running build_py
creating build/lib/nixos_rebuild
copying nixos_rebuild/services.py -> build/lib/nixos_rebuild
copying nixos_rebuild/nix.py -> build/lib/nixos_rebuild
copying nixos_rebuild/models.py -> build/lib/nixos_rebuild
copying nixos_rebuild/constants.py -> build/lib/nixos_rebuild
copying nixos_rebuild/tmpdir.py -> build/lib/nixos_rebuild
copying nixos_rebuild/__init__.py -> build/lib/nixos_rebuild
copying nixos_rebuild/utils.py -> build/lib/nixos_rebuild
copying nixos_rebuild/process.py -> build/lib/nixos_rebuild
copying nixos_rebuild/__main__.py -> build/lib/nixos_rebuild
running egg_info
writing nixos_rebuild_ng.egg-info/PKG-INFO
writing dependency_links to nixos_rebuild_ng.egg-info/dependency_links.txt
writing entry points to nixos_rebuild_ng.egg-info/entry_points.txt
writing top-level names to nixos_rebuild_ng.egg-info/top_level.txt
reading manifest file 'nixos_rebuild_ng.egg-info/SOURCES.txt'
writing manifest file 'nixos_rebuild_ng.egg-info/SOURCES.txt'
copying nixos_rebuild/repl.nix.template -> build/lib/nixos_rebuild
installing to build/bdist.macosx-11.3-x86_64/wheel
running install
running install_lib
creating build/bdist.macosx-11.3-x86_64/wheel
creating build/bdist.macosx-11.3-x86_64/wheel/nixos_rebuild
copying build/lib/nixos_rebuild/services.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/nix.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/models.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/constants.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/tmpdir.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/__init__.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/repl.nix.template -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/utils.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/process.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
copying build/lib/nixos_rebuild/__main__.py -> build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild
running install_egg_info
Copying nixos_rebuild_ng.egg-info to build/bdist.macosx-11.3-x86_64/wheel/./nixos_rebuild_ng-0.0.0-py3.13.egg-info
running install_scripts
creating build/bdist.macosx-11.3-x86_64/wheel/nixos_rebuild_ng-0.0.0.dist-info/WHEEL
creating '/private/tmp/nix-build-nixos-rebuild-ng-0.0.0.drv-0/src/dist/.tmp-3m8ptwzj/nixos_rebuild_ng-0.0.0-py3-none-any.whl' and adding 'build/bdist.macosx-11.3-x86_64/wheel' to it
adding 'nixos_rebuild/__init__.py'
adding 'nixos_rebuild/__main__.py'
adding 'nixos_rebuild/constants.py'
adding 'nixos_rebuild/models.py'
adding 'nixos_rebuild/nix.py'
adding 'nixos_rebuild/process.py'
adding 'nixos_rebuild/repl.nix.template'
adding 'nixos_rebuild/services.py'
adding 'nixos_rebuild/tmpdir.py'
adding 'nixos_rebuild/utils.py'
adding 'nixos_rebuild_ng-0.0.0.dist-info/METADATA'
adding 'nixos_rebuild_ng-0.0.0.dist-info/WHEEL'
adding 'nixos_rebuild_ng-0.0.0.dist-info/entry_points.txt'
adding 'nixos_rebuild_ng-0.0.0.dist-info/top_level.txt'
adding 'nixos_rebuild_ng-0.0.0.dist-info/RECORD'
removing build/bdist.macosx-11.3-x86_64/wheel
Successfully built nixos_rebuild_ng-0.0.0-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for nixos_rebuild_ng-0.0.0-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
Executing pypaInstallPhase
Successfully installed nixos_rebuild_ng-0.0.0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
checking for references to /private/tmp/nix-build-nixos-rebuild-ng-0.0.0.drv-0/ in /nix/store/z7nl9ii0dbcm9asx6s9lmi9xqvmvg2dx-nixos-rebuild-ng-0.0.0...
gzipping man pages under /nix/store/z7nl9ii0dbcm9asx6s9lmi9xqvmvg2dx-nixos-rebuild-ng-0.0.0/share/man/
patching script interpreter paths in /nix/store/z7nl9ii0dbcm9asx6s9lmi9xqvmvg2dx-nixos-rebuild-ng-0.0.0
stripping (with command strip and flags -S) in  /nix/store/z7nl9ii0dbcm9asx6s9lmi9xqvmvg2dx-nixos-rebuild-ng-0.0.0/lib /nix/store/z7nl9ii0dbcm9asx6s9lmi9xqvmvg2dx-nixos-rebuild-ng-0.0.0/bin
checking for references to /private/tmp/nix-build-nixos-rebuild-ng-0.0.0.drv-0/ in /nix/store/447njs0i4pb0y7r0xcky33lmq817s0xw-nixos-rebuild-ng-0.0.0-dist...
patching script interpreter paths in /nix/store/447njs0i4pb0y7r0xcky33lmq817s0xw-nixos-rebuild-ng-0.0.0-dist
Rewriting #!/nix/store/il3s6rjh49l9ccg1nlqm93w67lcnfmqn-python3-3.13.4/bin/python3.13 to #!/nix/store/il3s6rjh49l9ccg1nlqm93w67lcnfmqn-python3-3.13.4
wrapping `/nix/store/z7nl9ii0dbcm9asx6s9lmi9xqvmvg2dx-nixos-rebuild-ng-0.0.0/bin/nixos-rebuild-ng'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
Running phase: pythonRemoveBinBytecodePhase
Running phase: pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
Executing pytestCheckPhase
pytest flags: -m pytest -vv
============================= test session starts ==============================
platform darwin -- Python 3.13.4, pytest-8.3.5, pluggy-1.6.0 -- /nix/store/il3s6rjh49l9ccg1nlqm93w67lcnfmqn-python3-3.13.4/bin/python3.13
cachedir: .pytest_cache
rootdir: /private/tmp/nix-build-nixos-rebuild-ng-0.0.0.drv-0/src
configfile: pyproject.toml
collected 47 items                                                             

tests/test_main.py::test_parse_args PASSED                               [  2%]
tests/test_main.py::test_execute_nix_boot PASSED                         [  4%]
tests/test_main.py::test_execute_nix_build_vm PASSED                     [  6%]
tests/test_main.py::test_execute_nix_build_image_flake PASSED            [  8%]
tests/test_main.py::test_execute_nix_switch_flake PASSED                 [ 10%]
tests/test_main.py::test_execute_nix_switch_build_target_host PASSED     [ 12%]
tests/test_main.py::test_execute_nix_switch_flake_target_host PASSED     [ 14%]
tests/test_main.py::test_execute_nix_switch_flake_build_host PASSED      [ 17%]
tests/test_main.py::test_execute_switch_rollback PASSED                  [ 19%]
tests/test_main.py::test_execute_build PASSED                            [ 21%]
tests/test_main.py::test_execute_test_flake PASSED                       [ 23%]
tests/test_main.py::test_execute_test_rollback PASSED                    [ 25%]
tests/test_models.py::test_build_attr_from_arg PASSED                    [ 27%]
tests/test_models.py::test_build_attr_to_attr PASSED                     [ 29%]
tests/test_models.py::test_flake_parse PASSED                            [ 31%]
tests/test_models.py::test_flake_to_attr PASSED                          [ 34%]
tests/test_models.py::test_flake_from_arg FAILED                         [ 36%]
tests/test_models.py::test_profile_from_arg PASSED                       [ 38%]
tests/test_nix.py::test_build PASSED                                     [ 40%]
tests/test_nix.py::test_build_flake PASSED                               [ 42%]
tests/test_nix.py::test_build_remote PASSED                              [ 44%]
tests/test_nix.py::test_build_remote_flake PASSED                        [ 46%]
tests/test_nix.py::test_copy_closure PASSED                              [ 48%]
tests/test_nix.py::test_edit PASSED                                      [ 51%]
tests/test_nix.py::test_edit_flake PASSED                                [ 53%]
tests/test_nix.py::test_get_build_image_variants PASSED                  [ 55%]
tests/test_nix.py::test_get_build_image_variants_flake PASSED            [ 57%]
tests/test_nix.py::test_get_nixpkgs_rev PASSED                           [ 59%]
tests/test_nix.py::test_get_generations PASSED                           [ 61%]
tests/test_nix.py::test_get_generations_from_nix_env PASSED              [ 63%]
tests/test_nix.py::test_list_generations PASSED                          [ 65%]
tests/test_nix.py::test_repl PASSED                                      [ 68%]
tests/test_nix.py::test_repl_flake PASSED                                [ 70%]
tests/test_nix.py::test_rollback PASSED                                  [ 72%]
tests/test_nix.py::test_rollback_temporary_profile PASSED                [ 74%]
tests/test_nix.py::test_set_profile PASSED                               [ 76%]
tests/test_nix.py::test_switch_to_configuration_without_systemd_run PASSED [ 78%]
tests/test_nix.py::test_switch_to_configuration_with_systemd_run PASSED  [ 80%]
tests/test_nix.py::test_upgrade_channels PASSED                          [ 82%]
tests/test_process.py::test_run PASSED                                   [ 85%]
tests/test_process.py::test__kill_long_running_ssh_process PASSED        [ 87%]
tests/test_process.py::test_remote_from_name PASSED                      [ 89%]
tests/test_services.py::test_reexec PASSED                               [ 91%]
tests/test_services.py::test_reexec_flake PASSED                         [ 93%]
tests/test_utils.py::test_dict_to_flags PASSED                           [ 95%]
tests/test_utils.py::test_remap_dicts PASSED                             [ 97%]
tests/test_utils.py::test_tabulate PASSED                                [100%]

=================================== FAILURES ===================================
_____________________________ test_flake_from_arg ______________________________

mock_node = <function node at 0x10d20f600>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x10d1f4d60>
tmpdir = local('/private/tmp/nix-build-nixos-rebuild-ng-0.0.0.drv-0/pytest-of-_nixbld1/pytest-0/test_flake_from_arg0')

    @patch("platform.node", autospec=True)
    def test_flake_from_arg(
        mock_node: Mock, monkeypatch: MonkeyPatch, tmpdir: Path
    ) -> None:
        mock_node.return_value = "hostname"
    
        # Flake string
        assert m.Flake.from_arg("/path/to/flake#attr", None) == m.Flake(
            Path("/path/to/flake"), 'nixosConfigurations."attr"'
        )
    
        # False
        assert m.Flake.from_arg(False, None) is None
    
        # True
        with monkeypatch.context() as patch_context:
            patch_context.chdir(tmpdir)
            assert m.Flake.from_arg(True, None) == m.Flake(
                Path("."), 'nixosConfigurations."hostname"'
            )
    
        # None when we do not have /etc/nixos/flake.nix
        with patch(
            "pathlib.Path.exists",
            autospec=True,
            return_value=False,
        ):
            assert m.Flake.from_arg(None, None) is None
    
        # None when we have a file in /etc/nixos/flake.nix
        with (
            patch(
                "pathlib.Path.exists",
                autospec=True,
                return_value=True,
            ),
            patch(
                "pathlib.Path.is_symlink",
                autospec=True,
                return_value=False,
            ),
        ):
>           assert m.Flake.from_arg(None, None) == m.Flake(
                Path("/etc/nixos"), 'nixosConfigurations."hostname"'
            )
E           assert Flake(path=PosixPath('/private/etc/nixos'), attr='nixosConfigurations."hostname"') == Flake(path=PosixPath('/etc/nixos'), attr='nixosConfigurations."hostname"')
E             
E             Matching attributes:
E             ['attr']
E             Differing attributes:
E             ['path']
E             
E             Drill down into differing attribute path:
E               path: PosixPath('/private/etc/nixos') != PosixPath('/etc/nixos')

tests/test_models.py:116: AssertionError
=========================== short test summary info ============================
FAILED tests/test_models.py::test_flake_from_arg - assert Flake(path=PosixPath('/private/etc/nixos'), attr='nixosConfigurations."hostname"') == Flake(path=PosixPath('/etc/nixos'), attr='nixosConfigurations."hostname"')
  
  Matching attributes:
  ['attr']
  Differing attributes:
  ['path']
  
  Drill down into differing attribute path:
    path: PosixPath('/private/etc/nixos') != PosixPath('/etc/nixos')
========================= 1 failed, 46 passed in 0.45s =========================
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.29.1`
 - nixpkgs: `/nix/store/25wqidxgjxy2bii879wmkhfrwmiy0gjf-source`

### Notify maintainers

@thiagokokada

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ni/nixos-rebuild-ng/src/tests/test_models.py
+++ b/pkgs/by-name/ni/nixos-rebuild-ng/src/tests/test_models.py
@@ -108,9 +108,9 @@ def test_flake_from_arg(
             return_value=True,
         ),
         patch(
-            "pathlib.Path.is_symlink",
+            "pathlib.Path.resolve",
             autospec=True,
-            return_value=False,
+            return_value=Path("/etc/nixos/flake.nix"),
         ),
     ):
         assert m.Flake.from_arg(None, None) == m.Flake(
@@ -120,11 +120,6 @@ def test_flake_from_arg(
     with (
         patch(
             "pathlib.Path.exists",
-            autospec=True,
-            return_value=True,
-        ),
-        patch(
-            "pathlib.Path.is_symlink",
             autospec=True,
             return_value=True,
         ),

